cmd/gdeploy/main.go

```diff
@@ -165,14 +166,21 @@ func RequireAWSCredentials() cli.BeforeFunc {
 
 		stsClient := sts.NewFromConfig(cfg)
 		// Use the sts api to check if these credentials are valid
-		_, err = stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+		identity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
 		if err != nil {
 			var ae smithy.APIError
 			// the aws sdk doesn't seem to have a concrete type for ExpiredToken so instead we check the error code
 			if errors.As(err, &ae) && ae.ErrorCode() == "ExpiredToken" {
 				return clio.NewCLIError("AWS credentials are expired.", clio.LogMsg("Please export valid AWS credentials to run this command."))
 			}
-			return clio.NewCLIError("Failed to call AWS get caller identity", clio.DebugMsg(err.Error()))
+			return clio.NewCLIError("Failed to call AWS get caller identity. (maybe you need to assume a role first?)", clio.DebugMsg(err.Error()))
+		}
+
+		//allow the overwrite if overwrite is set
+		overwrite := c.Bool("overwrite")
```

Comment:
I don't think we need to have an override here at all, users should be deploying with the correct account every time.
You cannot do anything with the wrong account
```suggestion
```

Reply:
This was to allow for the scenario where you want to overwrite your config with a different account. If you had creds for a different account it would never let you override.

The only way would be to delete the config and re-init to bypass this I think

Reply:
In this case the customer would be creating a whole new deployment, they could change the account number manually in their config.

It better for us to help prevent this as much as possible.